extensions-core/datasketches/src/main/java/org/apache/druid/query/aggregation/datasketches/quantiles/DoublesSketchToHistogramPostAggregator.java

```diff
@@ -29,45 +29,68 @@
 import org.apache.druid.query.aggregation.PostAggregator;
 import org.apache.druid.query.cache.CacheKeyBuilder;
 
+import javax.annotation.Nullable;
 import java.util.Arrays;
 import java.util.Comparator;
 import java.util.Map;
 import java.util.Set;
 
 public class DoublesSketchToHistogramPostAggregator implements PostAggregator
 {
+  static final int DEFAULT_NUM_BINS = 10;
 
   private final String name;
   private final PostAggregator field;
   private final double[] splitPoints;
+  private final Integer numBins;
 
   @JsonCreator
   public DoublesSketchToHistogramPostAggregator(
       @JsonProperty("name") final String name,
       @JsonProperty("field") final PostAggregator field,
-      @JsonProperty("splitPoints") final double[] splitPoints)
+      @JsonProperty("splitPoints") @Nullable final double[] splitPoints,
+      @JsonProperty("numBins") @Nullable final Integer numBins)
   {
     this.name = Preconditions.checkNotNull(name, "name is null");
     this.field = Preconditions.checkNotNull(field, "field is null");
-    this.splitPoints = Preconditions.checkNotNull(splitPoints, "array of split points is null");
+    this.splitPoints = splitPoints;
+    this.numBins = numBins;
   }
 
   @Override
   public Object compute(final Map<String, Object> combinedAggregators)
   {
     final DoublesSketch sketch = (DoublesSketch) field.compute(combinedAggregators);
+    final int numBins = splitPoints != null ? splitPoints.length + 1 :
```

Comment:
Since `this.numBins` will be ignored if `splitPoints` is set, it'd be best to throw an error in the constructor if both are provided.  Could you please add one (an error message like `"Cannot accept both 'numBins' and 'splitPoints'"`).